bin/ch/MessageQueue.h

```diff
@@ -102,6 +102,18 @@ class SortedList
         }
     }
 
+    template <typename PredicateFn>
+    void RemoveAll(PredicateFn fn)
+    {
+        DListNode<T>* node = head;
+
+        while (node != nullptr)
+        {
+            fn(node->data);
+            node = node->next;
+        }
+    }
+
```

Comment:
nit: This function doesn't "remove" anything, shouldn't have name `RemoveAll`.

Reply:
hmm good point, I'll rename this.